vmdb/app/models/metric/ci_mixin/capture/openstack_base.rb

```diff
@@ -37,134 +37,364 @@ def perf_release_openstack
+    $log.debug "#{log_header} id:[#{name}] start_time: #{start_time}, end_time: #{end_time}"
+
+    counters = find_meter_counters(metric_capture_module, resource_filter, metadata_filter, log_header)
+
+    # TODO(lsmola) we can't be sure Ceilometer will be set with this value configured. If period of collecting the
+    # of the data will be bigger, we can have 'holes' in the 20s aligned data stream. So this value should be inferred
+    # from the data itself.
+    # For now hardcoding to 10 minutes which is Ceilometer default pipeline setting
+    data_collecting_period = 10.minutes
+
+    # We will have to account for the fact that each counter can be configured
+    # for individual capture intervals ... the out-of-box default is 10min
+    start_time -= data_collecting_period * 2
+
+    # Remove seconds from the start_time and end_time, so we always start at 00s and continue on aligned 20s steps
+    start_time -= start_time.sec
+    end_time -= end_time.sec
+
+    metrics_by_counter_name = {}
+    counters.each do |c|
+      metrics_by_counter_name[c["name"]] = collect_metrics_by_counter(c, start_time, end_time)
+    end
+
+    # TODO(lsmola) check that first statistic is already saved in the database. If not log.warn that there is hole
+    # in the data due to missing data in Ceilometer. Only if this is not the first metrics collection,
+
+    # TODO(lsmola) since it actually saves all counters at once, and we don't know if samples of all those counters
+    # will be aligned. We need to obtain the overlapping period of start and merge those samples together, so they
+    # are not rewritten by nil values of another collection period. This should probably go to saving algorithm
+
+    counter_values_by_ts = process_statistics(metric_capture_module, metrics_by_counter_name, data_collecting_period,
+                                              log_header)
+    counters_by_id              = {ems_ref => metric_capture_module::VIM_STYLE_COUNTERS}
+    counter_values_by_id_and_ts = {ems_ref => counter_values_by_ts}
+    return counters_by_id, counter_values_by_id_and_ts
+  end
+
+  def list_meters(filter)
+    counters, _ = Benchmark.realtime_block(:capture_counters) do
+      @perf_ems.list_meters([filter]).body
+    end
+    counters.each { |c| c[:instance_filter] = filter }
+    counters
+  end
+
+  #####################################################################################################################
+  # Private methods
+
+  private
 
-    instance_filter = resource_filter
+  def meter_names(metric_capture_module)
+    @meter_names ||= metric_capture_module::COUNTER_NAMES
+  end
+
+  def find_meter_counters(metric_capture_module, resource_filter, metadata_filter, log_header)
+    counters = list_resource_meters(resource_filter, log_header) + list_metadata_meters(metadata_filter, log_header)
+    counters.select { |c| meter_names(metric_capture_module).include? c["name"] }
```

Comment:
Minor, but can you put parens here on the `include?`?  When code is nested in a block like this, sometimes it gets confusing.